datafusion/optimizer/src/push_down_projection.rs

```diff
@@ -423,7 +426,93 @@ pub fn collect_projection_expr(projection: &Projection) -> HashMap<String, Expr>
         .collect::<HashMap<_, _>>()
 }
 
-// Get the projection exprs from columns in the order of the schema
+/// Accumulate the memory size of a data type measured in bits.
+///
+/// Nested types are traversed and increment `nesting` on every level.
```

Comment:
Can we add a comment saying that variable-sized types are estimated using some heuristics?

Reply:
Makes sense. Added a comment about variable sized types. Feel free to rephrase if you think something is missing.